litestar/logging/standard.py

```diff
@@ -18,7 +18,7 @@ class QueueListenerHandler(QueueHandler):
         """Configure queue listener and handler to support non-blocking logging configuration."""
 
         def __init__(self, handlers: list[Any] | None = None) -> None:
-            """Initialize `?QueueListenerHandler`.
+            """Initialize `QueueListenerHandler`.
```

Comment:
Was this a typo?

[The doc preview](https://litestar-org.github.io/litestar-docs-preview/3185/reference/logging/standard.html) seems to be broken, compared to the [official one](https://docs.litestar.dev/2/reference/logging/standard.html).

Reply:
Im looking into this

Reply:
@provinzkraut im not sure what makes it not load inside the preview, or if on next docs deploy the production docs would also stop loading?

I cant get them to load in any sort of configuration of Python version venvs and other means...
https://paste.pythondiscord.com/FKRQ is what I have for adding some notes to this, but i cant get our custom `QueueListenerHandler` to document itself in any way. Maybe it will be fine

Reply:
It's because this is hidden in an if block 🤷

Reply:
@provinzkraut the `if` block is already on the `main` branch and the doc is correct? 🤔

Reply:
But the docs build on `3.12`. For `3.12`, the `QueueListenerHandler` isn't defined. This is a general issue and a breaking change by the way. The previous version had `QueueListenerHandler` defined in both branches.

Reply:
@provinzkraut I guess I understand better.

But it means [the official doc](https://docs.litestar.dev/latest/reference/logging/standard.html) has been generated with Python < 3.12 since it refers to the former `QueueListenerHandler` class (not to the aliased one), right?

Or does it mean the official doc is not up-to-date (not generated/published recently)?

While I understand this is a breaking change at the API level, I'm not sure what I should do next. Any advice?

Reply:
I'm not 100% sure, but I think the issue is that the `LoggingQueueListener` isn't in `__all__` and what *is* in `__all__` (`QueueListenerHandler`), isn't available on < 3.12. My guess would be that if you rename ``LoggingQueueListener` > `QueueListenerHandler`, it should work. If not, then I'm wrong and the issue is something else

Reply:
You're right about `__all__`.

That's said, renaiming `LoggingQueueListener` to `QueueListenerHandler` is not a good solution IMO, because this is a `QueueListener` object and not a `logging.Handler` one. I think naming is important. Also, the constructor args are not the same and I don't think I can change that... So it would be confusing, and the API doc would be incorrect.

I could keep `QueueListenerHandler = QueueHandler`, but it won't help to get `QueueListenerHandler` fully documented (for Python < 3.12).

I could remove the `if` condition, it shouldn't be a problem (technically). Even if it cannot be used with Python >= 3.12, it can be documented. Or I can keep the `if` condition and redefine `QueueListenerHandler` with the same constructor signature, and raise an exception (or a warning) stating it cannot be used with Python > 3.12.

I'm not sure how to manage this at the project level.

I tried to take a second look to check if we could keep `QueueListenerHandler` around. 👀 And from what I've understood, the signature has to be changed so it would break the API anyway. This is because the handler is instanciated with a single argument, which is the queue (and not `handlers: list[Any] | None`): https://github.com/python/cpython/blob/v3.12.2/Lib/logging/config.py#L744 -- `_configure_queue_handler` has been introduced in Python 3.12.

Reply:
I think for now we'll have to just keep the `QueueListenerHandler = QueueHandler` around somehow. `QueueListenerHandler` *and* `LoggingQueueListener` need to always be defined, no matter what Python version is being used. They should also be both in `_all__`. 

> I could remove the if condition, it shouldn't be a problem (technically). Even if it cannot be used with Python >= 3.12, it can be documented.

I think that's the best approach for now.

Reply:
Sorry if all of this is confusing, but removing the `if` statement AND keeping `QueueListenerHandler = QueueHandler` won't work. :)

To clarify (and if I'm not mistaken):

Currently (`main` branch):
⛔️ The default `queue_listener` handler is broken on Python 3.12
⛔️ Generating the API doc shows `QueueListenerHandler` as an alias of `logging.QueueHandler`, which is not the case for Python < 3.12
⚠️ Official API doc still shows the previous implementation (Python < 3.12) for an unknown reason (?)
⛔️ When upgrading to Python 3.12, the constructor signature of`QueueListenerHandler` is not the same anymore, which implies a breaking change for any "custom" usage of `QueueListenerHandler`

Keeping `QueueListenerHandler = QueueHandler` won't help to get a correct API doc for Python < 3.12.

I guess removing the `if` statement seems better:
✅ The default `queue_listener` handler will work out of the box
✅ API doc will show the informations for `QueueListenerHandler` (and `LoggingQueueListener`)
✅ The fact it doesn't work with Python >= 3.12 can be documented
✅ `QueueListenerHandler` will be able to be deprecated later (maybe we could/should add a warning right now?)
⛔️ Any usage relying on `QueueListenerHandler = QueueHandler` will break (that's said, as of today, the official API doc doesn't mention this aliasing on Python >= 3.12) -- I guess it might be a problem mostly if some custom configs have been put in place because it was not working out of the box with Python 3.12.
⛔️ Usage of `QueueListenerHandler` will break when upgrading to Python 3.12 for any custom implementation using it. But it can be documented. And using `QueueListenerHandler = QueueHandler` wouldn't help, because the constructor signatures are not the same (as stated before).

In all cases, there is a breaking change we can't avoid due to Python 3.12. So the best we can do is making it clear. Also, I'm not sure anyone is using `QueueListenerHandler` outside of the `default_handlers` provided, so the change might be minimal (that's said, I'm new to Litestar).

Reply:
@provinzkraut I updated the PR, preview doc is available [here](https://litestar-org.github.io/litestar-docs-preview/3185/reference/logging/standard.html).

Any recommendation?

Reply:
Thanks for the great summary @jderrien! The current implementation seems like the best solution to me. Let's just keep it like that.